components/timezone/src/types.rs

```diff
@@ -123,3 +126,51 @@ impl FromStr for GmtOffset {
         Self::try_new(seconds)
     }
 }
+
+/// A time variant, e.g. "daylight" or "standard"
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash, ULE)]
+#[repr(transparent)]
+#[cfg_attr(feature = "datagen", derive(serde::Serialize))]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[allow(clippy::exhaustive_structs)] // newtype
+pub struct TimeVariant(pub TinyStr8);
```

Comment:
Thought: We should consider making this a `TinyStr4`

Reply:
and shortening the string? could work!

Reply:
`"std"` and `"day"` seem nice (or `"dst"`)

Reply:
`"std"` and `"dst"` sgtm. And for a future Ramadan variant we could do `"rmn"` or something.

Maybe even make it a `TinyAsciiStr<3>` :)

Reply:
https://github.com/unicode-org/icu4x/issues/2292